src/fidesops/ops/service/email/email_dispatch_service.py

```diff
@@ -16,30 +16,64 @@
     EmailServiceType,
     SubjectIdentityVerificationBodyParams,
 )
+from fidesops.ops.tasks import DatabaseTask, celery_app
 from fidesops.ops.util.logger import Pii
 
 logger = logging.getLogger(__name__)
 
 
+class InvalidBodyParams(Exception):
+    """An exception thrown when passed email body data does not match an allowed schema"""
+
+
+def _validate_email_body_params(
+    email_body_params: Dict[str, Union[str, int]]
+) -> SubjectIdentityVerificationBodyParams:
+    """Raises an exception if `email_body_params` doesn't fit one of the allowed schemas"""
+    email_body_schema_allowlist = [SubjectIdentityVerificationBodyParams]
+    for schema in email_body_schema_allowlist:
```

Comment:
This feels like something that would be built-in.

Maybe creating an EmailBodyParams class and setting `Config.SmartUnion` might work, or there may be other built-in options.

https://pydantic-docs.helpmanual.io/usage/model_config/#smart-union